packages/instantsearch.js/src/middlewares/__tests__/createInsightsMiddleware.ts

```diff
@@ -581,6 +569,74 @@ See documentation: https://www.algolia.com/doc/guides/building-search-ui/going-f
         ]
       `);
     });
+
+    it('does not call `init` when default middleware is used', () => {
```

Comment:
This assumes the "default" middleware always uses a version that's able not to `init`, which might not be the case if Search Insights is already present but using an older v2 version, no?

Reply:
good point, maybe it should be "shouldAddScript" as the condition instead?

Reply:
Should we update the condition to something like this:
```
if (!$$internal || insightsInitParams || !isModernInsightsClient(insightsClient)) {
  // ...
}
```

Reply:
@FabienMotte Would `isModernInsightsClient` use the `version`?

Reply:
Yes, the function already exists: https://github.com/algolia/instantsearch/blob/52f54bfb408c165f07184252f3737853b2c98a6c/packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts#L315

Reply:
Yeah, so the only scenarios when `init` should be called are **when the user provides `insightsInitParams`** _or_ **when the Insights version doesn't support not calling `init`**.

But the fact that it's internal or not doesn't count here, am I correct?

Reply:
I'd also clarify the condition, because `isModernInsightsClient` doesn't tell us what it actually supports.

Reply:
WDYT of `supportsOptionalInitMethod(insightsClient)`?